includes/options/class-amp-options-manager.php

```diff
@@ -199,6 +200,12 @@ public static function validate_options( $new_options ) {
 		// Store the current version with the options so we know the format.
 		$options['version'] = AMP__VERSION;
 
+		// Handle the caching option.
+		$options['enable_response_caching'] = ! empty( $new_options['enable_response_caching'] );
+		if ( $options['enable_response_caching'] ) {
+			do_action( 'amp_reenable_response_cache' );
```

Comment:
Wouldn't it not be better to call the method here rather than indirectly trigger the action that the method is is connected to? Is there any utility for the `amp_reenable_response_cache` action generally?

Reply:
Upon review, you're right.  There is not need for a separation action here.  Switching to call direct.